src/xunit.execution/Sdk/DisposalTracker.cs

```diff
@@ -23,8 +23,10 @@ public void Add(IDisposable disposable)
         /// <inheritdoc/>
         public void Dispose()
         {
-            foreach (var disposable in toDispose)
-                disposable.Dispose();
+            for (int i = 0; i < toDispose.Count; ++i)
```

Comment:
```suggestion
            for (int i = toDispose.Count; i > 0; --i)
```
to avoid fetching the count (which stack.Count is changing every iteration) [example run](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEUCuA7AHwAEAmARgFgAoIgBgAIiyUBua6gNwEMp6F6AvPTwwA7vQDKGLmADWAHgCWeDAD4AFAEo2VaggB0ABRwBnABbqy2vUdMWS1qgePn1AZkfUAZtHXKM9IqCfPoAwhD4GCyB9Kr0tNFwcIqa1PTpjGQAnOrOEAAOWo5AA)

Reply:
This also would've failed to dispose everything, because the count was changing (it would've only disposed roughly half the items).